### PR TITLE
Erichlf torrentday

### LINKED ISSUE #2272: TorrentDay No Results Found (Again)
html.parser is unstable under python2, so torrentday plugin seems to not be able to parse properly resulting in No Results Found.

### LINKED ISSUE #2273: Torrentday Seeders and Leechers Causes an Error.
File "/usr/local/flexget/env/lib/python2.7/site-packages/flexget/plugins/sites/torrentday.py", line 174, in search
    entry['torrent_seeds'] = int(seeders.contents[0].replace(',', ''))
  File "/usr/local/flexget/env/lib/python2.7/site-packages/bs4/element.py", line 1807, in __getattr__
    "ResultSet object has no attribute '%s'. You're probably treating a list of items like a single item. Did you call find_all() when you meant to call find()?" % key
AttributeError: ResultSet object has no attribute 'contents'. You're probably treating a list of items like a single item. Did you call find_all() when you meant to call find()?
2018-12-08 16:01 WARNING  task          AVI2MKV         Aborting task (plugin: discover)
2018-12-08 16:01 DEBUG    task_queue                    task AVI2MKV aborted: TaskAbort(reason=BUG: Unhandled error in plugin discover: ResultSet object has no attribute 'contents'. You're probably treating a list of items like a single item. Did you call find_all() when you meant to call find()?, silent=False)

### PR DESCRIPTION
**Pull request by erichlf**
### Motivation for changes:
torrentday webpages seemed to be malformed. To address this issue one can skip over that portion of the webpage.

Additionally, the seeders and leechers portion of td plugin results in an error, since it is using find_all when it should use find.

### Detailed changes:
- add  hack to bypass malformed portion of html
- switch from find_all to find for seeders and leechers

### Addressed issues:
- Fixes #2272
- Fixes #2273

### CODE REVIEW

**Comment by cvium** [comment]
You also have some conflicts so you probably need to pull the latest

File: flexget/plugins/sites/torrentday.py [hunk 1]
```diff
@@ -123,30 +123,44 @@ def search(self, task, entry, config=None):
...
             cookies = { 'uid': config['uid'], 'pass': config['passkey'], '__cfduid': config['cfduid'] }
 
             try:
                 page = requests.get(url, params=params, cookies=cookies).content
             except RequestException as e:
-                raise PluginError('Could not connect to torrentday: %s' % e)
+                raise PluginError('Could not connect to torrentday: {}'.format(e))
 
-            soup = get_soup(page)
+            soup = get_soup(page, "lxml")
```
[the remark this answers is not in the document]
Comment (erichlf) [reply flexget/plugins/sites/torrentday.py]: Don't know why it fixed the issue then.

**Comment by gazpachoking** [comment]
The `find` vs `find_all` bit looks good. I'm not so sure about introducing lxml as a dep though. html5lib doesn't work even if you split out the table find bit like you did here?

**Comment by erichlf** [comment]
I get inconsistent results using the default. The same search will sometimes give no results and sometimes give results, but mostly no results.

**Comment by erichlf** [comment]
Okay, so something else is going on here, because I get the same problem using LXML. I will have to dig deeper.

**Comment by erichlf** [comment]
Removed lxml from get_soup and added a hack to skip over malformed html.

**Pull request merged**